From af9cfeb5f94d9cd4f2989fc8e111c33208494ba4 Mon Sep 17 00:00:00 2001
From: Yaroslav Halchenko <debian@onerussian.com>
Date: Wed, 5 Dec 2012 11:51:18 -0500
Subject: [PATCH] fix analysis of entry to be a weakref (typo introduced in
 d96dfd)

Originally reported in Debian http://bugs.debian.org/cgi-bin/bugreport.cgi\?bug\=692313
Lead to crash with python-dbg:

  python-dbg: ../Modules/gcmodule.c:366: visit_decref: Assertion 'gc->gc.gc_refs != 0' failed
---
 Cython/Compiler/Symtab.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Cython/Compiler/Symtab.py b/Cython/Compiler/Symtab.py
index 4631d73..6ed9479 100644
--- a/Cython/Compiler/Symtab.py
+++ b/Cython/Compiler/Symtab.py
@@ -802,7 +802,7 @@ class Scope(object):
 
         for entry in self.var_entries:
             if entry.type.is_pyobject:
-                if include_weakref or entry.name != "weakref":
+                if include_weakref or entry.name != "__weakref__":
                     py_attrs.append(entry)
             elif entry.type == PyrexTypes.c_py_buffer_type:
                 py_buffers.append(entry)
-- 
1.8.0

